doc/tutorials/crypto.md

```diff
@@ -0,0 +1,106 @@
+# Cryptographic Methods in Elektra
+
+Elektra can protect the following aspects of your configuration:
+
+1. confidentiality, and
+2. integrity.
+
+Elektra provides two plugins to achieve this protection:
+
+1. `crypto`, and
+2. `fcrypt`.
+
+## Configuration File Encryption/Decryption
+
+The `fcrypt` plugin enables the encryption and decryption of entire configuration files, thus protecting the confidentiality of the configuration values.
+`fcrypt` utilizes GnuPG (GPG) for all cryptographic operations.
+The GPG key, which is used for encryption and decryption, is specified in the backend configuration under `encrypt/key`.
+You MUST be in possesion of the private key, otherwise `fcrypt` will deny all operations.
+
+Let's assume your GPG private key has the ID `DDEBEF9EE2DC931701338212DAF635B17F230E8D`.
+If you want to encrypt your configuration file `test.ini` you can mount the backend under `user/test` like this:
+
+	kdb mount test.ini user/test fcrypt "encrypt/key=DDEBEF9EE2DC931701338212DAF635B17F230E8D" ini
+
+As a result the file `test.ini` is encrypted using GnuPG.
+`fcrypt` will call the `gpg2` or `gpg` binary as follows:
+
+	gpg2 -o test.ini -a -r DDEBEF9EE2DC931701338212DAF635B17F230E8D -e test.ini.tmp
+
+Note that `test.ini` can not only be decrypted by Elektra, but it is also possible to decrypt it with GnuPG directly, as long as you are in possesion of the private key.
+
+## Configuration File Signatures
+
+`fcrypt` also offers the option to sign and verify configuration files, thus protecting the integrity of the configuration values.
+If `sign/key` is specified in the backend configuration, `fcrypt` will forward the key ID to be used for signing the configuration file.
+
+An example backend configuration is given as follows:
+
+	kdb mount test.ini user/test fcrypt "sign/key=DDEBEF9EE2DC931701338212DAF635B17F230E8D" ini
+	
+As a result the file `test.ini` will be signed using GPG.
+`fcrypt` will call the `gpg2` or `gpg` binary as follows:
+
+	gpg2 -o test.ini -a -u DDEBEF9EE2DC931701338212DAF635B17F230E8D -r DDEBEF9EE2DC931701338212DAF635B17F230E8D -s test.ini.tmp
+
+If `test.ini` is modified, all following calls of `kdb get` will fail with an error message stating that the signature of the file could not be verified.
+
+### Combining Signatures and Encryption
+
+The options `sign/key` and `encrypt/key` can be combined together, resulting in configuration files, that are signed and encrypted.
+
+## Configuration Value Encryption/Decryption
+
+The compilation variants of the `crypto` plugin:
```

Comment:
Nobody knows what compilation variants are (except you :smile:). Either link to it or even better: simply say that there are 3 different providers.

But actually even this might be irrelevant. You should also mention here which the recommended provider is. (the fastest). Maybe even update the contract (README.md, clause infos/status) to reflect which crypto compilation variants are recommended.